src/udev/udev-builtin-net_id.c

```diff
@@ -414,21 +428,26 @@ static int names_ccw(struct  udev_device *dev, struct netnames *names) {
         struct udev_device *cdev;
         const char *bus_id;
         size_t bus_id_len;
+        size_t bus_id_start;
         int rc;
+        char *subsys;
 
         assert(dev);
         assert(names);
 
         /* Retrieve the associated CCW device */
         cdev = udev_device_get_parent(dev);
+        /* skip virtio subsystem if present */
+        cdev = skip_virtio(cdev);
         if (!cdev)
                 return -ENOENT;
 
-        /* Network devices are always grouped CCW devices */
-        if (!streq_ptr("ccwgroup", udev_device_get_subsystem(cdev)))
+        /* Network devices are either single or grouped CCW devices */
+        subsys = udev_device_get_subsystem(cdev);
+        if (!STR_IN_SET(subsys, "ccwgroup", "ccw"))
```

Comment:
hmm, so the old code actually used streq_ptr(), i.e. was OK with u_d_g_s() returning NULL. I figure the new code should deal with that too, hence STRPTR_IN_SET should be used, instead of STR_IN_SET... Sorry for not noticing this earlier!